src/components/MyAccount/NewSettings/UsernameForm.tsx

```diff
@@ -0,0 +1,211 @@
+  FAILURE:
+    "Your changes could not be saved. Please try again or contact us for assistance. ",
+  SUCCESS: "Your changes were saved.",
+}
+
+interface UsernameFormProps {
+  patron: Patron
+  usernameState
+}
+
+const UsernameForm = ({ patron, usernameState }: UsernameFormProps) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+  const [error, setError] = useState(false)
+  const [input, setInput] = useState(patron.username || "")
+  const [tempInput, setTempInput] = useState(input)
+
+  const { setUsernameStatus, setUsernameStatusMessage } = usernameState
+
+  const validateUsername = (username: string | null) => {
+    if (username === null) return false
+    if (username === "") return true
+    const usernameRegex = /^[a-zA-Z0-9]{5,15}$/
+    return usernameRegex.test(username)
+  }
+
+  const cancelEditing = () => {
+    setTempInput(input)
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const handleInputChange = (e) => {
+    const { value } = e.target
+    setTempInput(value)
+    if (!validateUsername(value)) {
+      setError(true)
+    } else {
+      setError(false)
+    }
+  }
+
+  const submitInput = async () => {
+    setIsLoading(true)
+    setIsEditing(false)
+    setUsernameStatus("")
+    const submissionInput = tempInput === null ? "" : tempInput
+    try {
+      const response = await fetch(
+        `${BASE_URL}/api/account/username/${patron.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ username: submissionInput }),
+        }
+      )
+      const responseMessage = await response.json()
+      if (responseMessage !== "Username taken" && response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setUsernameStatus("success")
+        setInput(submissionInput)
+        setTempInput(submissionInput)
+      } else {
+        setUsernameStatus("failure")
+        setUsernameStatusMessage(
+          responseMessage === "Username taken"
+            ? usernameStatusMessages.USERNAME_FAILURE
+            : usernameStatusMessages.FAILURE
+        )
+        setTempInput(input)
+      }
+    } catch (error) {
+      setUsernameStatus("failure")
+      setUsernameStatusMessage(usernameStatusMessages.FAILURE)
+      console.error("Error submitting username:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <Flex
+      flexDir={{ base: "column", lg: "row" }}
+      alignItems="flex-start"
+      width="100%"
+    >
+      <Flex gap="xs" marginRight="140px" paddingTop="xs" alignItems="center">
+        <Icon name="actionIdentity" size="large" />
+        <Text size="body1" sx={{ fontWeight: "500", marginBottom: 0 }}>
+          Username
+        </Text>
+      </Flex>
+
+      {isLoading ? (
```

Comment:
Similar request here. Can you replace this nested ternary with a conditional outside of the return statement so it's a easier to scan which components get rendered when?

Reply:
Again lmk if this is better/ more readable

Reply:
Same note here - the extraction is good, but the nested ternary itself is hard to read. Can you replace it?